### PR TITLE
command for copying data in cache

### PR DESCRIPTION
**Pull request by vlimant**
with a couple of options in command line
```
--cache-data /tmp/vlimant
--copy-command "cp {} {}"

```

which can also be driven with
```
export DATA_CACHE=/tmp/vlimant
export DATA_COPY_COMMAND="s3cmd get s3://gan-bucket/{} {}"
```

### CODE REVIEW

**Comment by vlimant** [comment]
@svalleco looks like it preserve the s3 functionality

File: nnlo/train/data.py [hunk 1]
```diff
@@ -90,25 +93,28 @@ def set_file_names(self, file_names):
...
-                          relocated.append( fn ) ## use the initial one
+                    if self.copy_command:
+                        cmd = self.copy_command.format( fn, relocate )
+                        if os.system(cmd)==0:
+                            relocated.append( relocate )
+                        else:
+                            logging.error("was unable to copy the file with {}".format( cmd ))
+                            relocated.append( fn ) ## use the initial one
+                    else:
+                        ## admitedly, this is redundant, as copy_command could default to "cp {} {}"
```
Comment (vloncar) [inline nnlo/train/data.py]: We can set the default `copy_command` in `Data`s `__init__` to get rid of this if statement.

File: nnlo/train/data.py [hunk 2]
```diff
@@ -90,25 +93,28 @@ def set_file_names(self, file_names):
...
+                            relocated.append( fn ) ## use the initial one
                 else:
                     relocated.append( relocate )
                         
             self.file_names = relocated
         else:
             self.file_names = file_names
-            
+
+        self.file_names = list(filter(None, self.file_names))
```
Comment (vloncar) [inline nnlo/train/data.py:109]: I don't see why we need to filter out the bad names (unless a list with None filenames was initially provided)?

Comment (vlimant) [reply nnlo/train/data.py:109]: that's a simple protection

Comment (vloncar) [reply nnlo/train/data.py:109]: `mkdir -p` a on line 90 would also be a good simple protection ;-)

File: nnlo/train/data.py (hunk 1 above)
Comment (vlimant) [reply nnlo/train/data.py]: yup

File: nnlo/train/data.py (hunk 2 above)
Comment (vlimant) [reply nnlo/train/data.py:109]: 15fdca4

**Review by vloncar — APPROVED** [review]
Looks good!

**Pull request merged**